### DISCUSSION TITLE
Is it possible to disable mouse reporting entirely?

### DISCUSSION THREAD

**Discussion by signed-log** [wezterm/wezterm#2365] [Q&A] [answered]
I am trying to disable mouse reporting support (when a shell app supports it like `vim`) outright and only keep a behaviour of not passing mouse inputs

But the docs only mentions *giving up* the non-reporting mode and not on the reporting mode (by not setting any modifier)

> The [bypass_mouse_reporting_modifiers](https://wezfurlong.org/wezterm/config/lua/config/bypass_mouse_reporting_modifiers.html) option allows you to specify an alternative set of modifiers to use for bypassing mouse reporting capture.

Is it possible to do so?

**Comment by wez** [comment, marked as answer]
No, it's not possible to turn it off. What's your higher level goal? Seems like you might want to configure vim to disable its use of the mouse?

**Reply by signed-log** [reply]
Thanks for your answer @wez 

I was hoping for a global way to disable terminal mouse support like most other terminal emulators allow you to do, but I can settle on doing it on a app-by-app basis as this is not possible here.

Thanks again

**Reply by gwillen** [reply]
@wez Would you accept a patch for a config option to disable mouse reporting? I'm currently in the market for a new terminal, specifically with the purpose of finding one that allows me to disable mouse reporting, since gnome-terminal does not offer this and has explicitly stated they will not.

My higher-level goal is that I do not want to have to look up how to individually disable use of the mouse in each separate app I'm using. Vim and claude code are the current main offenders, but every time a new app surprises me with this "feature" it just annoys me yet again. I'm using a terminal because I want it to behave like a terminal. When I use my mouse in a terminal, it is ALWAYS because I want to select text, and NEVER because I want whatever stupid thing the app developer came up with.

(Alternatively, perhaps even better: a config option to swap default behavior, so that text selection happens by default, and a modifier key is required to do mouse reporting instead? Either way, I'm happy to provide a pull request, I just want to know whether you'd even consider accepting one. Thanks!)

**Reply by bew** [reply]
Please open a dedicated issue for this, I understand the wish for determinism and low surprises from interactions with terminal programs.
I think a config option to swap the modifier behavior (allow reporting, not prevent it) is feasible, let's talk about it in your issue!

---
And avoid resurrecting old discussions like this ^^'

**Comment by konrad-npx** [comment]
I did `config.bypass_mouse_reporting_modifiers = ''` and it doesn't seem to report mouse events anymore (which is good but different from what you described).
